transport/http2_client.go

```diff
@@ -551,6 +551,16 @@ func (t *http2Client) CloseStream(s *Stream, err error) {
 		t.mu.Unlock()
 		return
 	}
+	if err != nil {
+		// notify in-flight streams, before the deletion
+		s.write(recvMsg{err: err})
+
+		if _, ok := err.(StreamError); ok {
+			// trigger server to send GoAway
+			s.rstStream = true
+			s.rstError = http2.ErrCodeCancel
```

Comment:
This doesn't trigger the server to send GoAway. Moreover, what's the rationale behind triggering it? It is a connection level concept; closing of a stream with some error shouldn't trigger closing of a connection, right?

We liked your original change, any specific reasons for updating it?